lib/queue.rb

```diff
@@ -22,10 +45,13 @@ def size
   end
 
   def empty?
-    raise NotImplementedError, "Not yet implemented"
+    if @front == @back
+      return true
+    end
   end
 
   def to_s
+    @store.delete(nil)
```

Comment:
This will shrink the internal array.  It's both an expensive operation and it will also not maintain the order of elements in the Queue.